router/middleware/perm/perm.go

```diff
@@ -437,8 +438,23 @@ func MustRead() gin.HandlerFunc {
 			return
 		}
 
+		// return if request is from worker with build token access
+		if strings.EqualFold(cl.TokenType, constants.WorkerBuildTokenType) {
+			b := build.Retrieve(c)
```

Comment:
MustRead is sort of resource agnostic. `org` and `repo` are typically present on any endpoint today, especially one that needs perm check. this addition assumes we are on a route that did `build.Establish()`. should there be additional checks or a different solution to account for this?

Reply:
It only assumes we're on a route that did `build.Establish()` if the token type is `WorkerBuild`, which is an accurate assumption to make. If the build token is used on an endpoint that does not include the `build.Establish()` piece, then it will fail without panic. 

E.g. tried using a build token for GET `api/v1/repos/foo/bar` and was met with 

```
subject vela-worker does not have 'read' permissions for repo foo/bar
```

Could be a more descriptive error, but at the very least it does not panic

Reply:
Agreed on the error message. My concern was two-fold. One was the thought that build would provide `nil`, but since we're using getters we should be shielded from not `nil` checking here. The other concern was more about stuffing route/token specific logic in these, otherwise, route and token agnostic methods. I can see that getting unwieldy if it gets out of hand.